File: src/test/resources/mark_java/const.java
public class CT {

    public static final int CONSTANT = 3;

    enum Color
    {
        RED, GREEN, BLUE;
    }

    public static void main(String[] args){
        CT c = new CT();

        c.foo(Color.RED);

        c.bar(CT.CONSTANT);

        c.bar(3);
    }

    private void bar(int constant) {
    }

    private void foo(Color red) {
    }
}

File: src/test/resources/mark_cpp/const.cpp
#include <iostream>


enum Color{
    red=5,
    black
};


class Test {

public:
	static const int CONSTANT = 8;


    int foo(Color c) {
        return c;
    }

    int bar(int i) {
    	return i;
    }
};


int main() {
  int foo;
  Test t;

  t.foo(Color::red);
  t.bar(Test::CONSTANT);
  t.bar(8);
  t.bar(7);
}

